package uz.pdp;

import org.springframework.context.annotation.AnnotationConfigApplicationContext;
import org.springframework.expression.Expression;
import org.springframework.expression.ExpressionParser;
import org.springframework.expression.spel.standard.SpelExpressionParser;

public class Main {
    public static void main(String[] args) {
        ExpressionParser parser = new SpelExpressionParser();
//        Expression expression = parser.parseExpression("'jon ' + 'doe'");
        Expression expression = parser.parseExpression("('jon ' + 'doe').toUpperCase()");
        Object value = expression.getValue();
        System.out.println(value);

        AnnotationConfigApplicationContext context =
                new AnnotationConfigApplicationContext(MyConfig.class);
        User user = context.getBean(User.class);
        System.out.println(user);

    }
}